src/lib/pickles/wrap.ml

```diff
@@ -72,560 +72,218 @@ let combined_inner_product (type actual_proofs_verified) ~env ~domain ~ft_eval1
-    ; zeta = scalar_chal O.zeta
-    ; joint_combiner =
-        (* TODO: Needs to be changed when lookups are fully implemented *)
-        Option.map (O.joint_combiner_chal o)
-          ~f:(Scalar_challenge.map ~f:Challenge.Constant.of_tick_field)
-    ; feature_flags = actual_feature_flags
+        , shifted_tick_field
+        , scalar_challenge_constant Import.Bulletproof_challenge.t
+          Import.Types.Step_bp_vec.t
+        , Import.Branch_data.t )
+        Import.Types.Wrap.Proof_state.Deferred_values.t
     }
-  in
-  let r = scalar_chal O.u in
-  let xi = scalar_chal O.v in
-  let module As_field = struct
-    let to_field =
-      SC.to_field_constant
-        (module Tick.Field)
-        ~endo:Endo.Wrap_inner_curve.scalar
-
-    let r = to_field r
-
-    let xi = to_field xi
-
-    let zeta = to_field plonk0.zeta
-
-    let alpha = to_field plonk0.alpha
-
-    let joint_combiner = Option.map ~f:to_field plonk0.joint_combiner
-  end in
-  let domain = Domain.Pow_2_roots_of_unity step_vk.domain.log_size_of_group in
-  let zetaw = Tick.Field.mul As_field.zeta step_vk.domain.group_gen in
-  let tick_plonk_minimal =
-    { plonk0 with
-      zeta = As_field.zeta
-    ; alpha = As_field.alpha
-    ; joint_combiner = As_field.joint_combiner
-    }
-  in
-  let tick_combined_evals =
-    Plonk_checks.evals_of_split_evals
-      (module Tick.Field)
-      proof.proof.openings.evals ~rounds:(Nat.to_int Tick.Rounds.n)
-      ~zeta:As_field.zeta ~zetaw
-    |> Plonk_types.Evals.to_in_circuit
-  in
-  let tick_domain =
-    Plonk_checks.domain
-      (module Tick.Field)
-      domain ~shifts:Common.tick_shifts
-      ~domain_generator:Backend.Tick.Field.domain_generator
-  in
-  let tick_env =
-    let module Env_bool = struct
-      type t = bool
-
-      let true_ = true
-
-      let false_ = false
-
-      let ( &&& ) = ( && )
 
-      let ( ||| ) = ( || )
-
-      let any = List.exists ~f:Fn.id
-    end in
-    let module Env_field = struct
-      include Tick.Field
-
-      type bool = Env_bool.t
-
-      let if_ (b : bool) ~then_ ~else_ = if b then then_ () else else_ ()
-    end in
-    Plonk_checks.scalars_env
-      (module Env_bool)
-      (module Env_field)
-      ~endo:Endo.Step_inner_curve.base ~mds:Tick_field_sponge.params.mds
-      ~zk_rows:step_vk.zk_rows ~srs_length_log2:Common.Max_degree.step_log2
-      ~field_of_hex:(fun s ->
-        Kimchi_pasta.Pasta.Bigint256.of_hex_string s
-        |> Kimchi_pasta.Pasta.Fp.of_bigint )
-      ~domain:tick_domain tick_plonk_minimal tick_combined_evals
-  in
-  let plonk =
-    let module Field = struct
-      include Tick.Field
-    end in
-    Type1.derive_plonk
-      (module Field)
-      ~shift:Shifts.tick1 ~env:tick_env tick_plonk_minimal tick_combined_evals
-  and new_bulletproof_challenges, b =
-    let prechals =
-      Array.map (O.opening_prechallenges o) ~f:(fun x ->
-          Scalar_challenge.map ~f:Challenge.Constant.of_tick_field x )
+  let deferred_values (type n)
```

Comment:
It's probably not, otherwise the CI wouldn't pass (or `mina` wouldn't compile)

Reply:
I just checked: it is because you `include For_tests_only` on the top level here, just before `wrap`. 
https://github.com/MinaProtocol/mina/pull/14637/files#diff-9e002804af8996a79819ae0cf83ada29f888c4232560ef99a989a4f1001c101aR286

Maybe it makes sense to move `deferred_values` outside of `For_tests_only`?

Reply:
Got you. Well we still need to export it *and* mark that's only for testing purposes. So either we do 

```
let f = ....

module For_tests = struct let f = f end 
```

or 

```
module For_tests = struct let f = ... end 
include For_tests
```
I kind of liked the second way better since it would extend to any further addition of function.